### PR TITLE
!superstarify default time

### LINKED ISSUE #1515: Default duration for !superstar command if no duration is specified
## Context
The `!superstar` command currently requires a duration to be specified when invoked. 

Often we want to change someone's nickname from something very noisy or that takes up a lot of room until the user can respond and change it themselves. The duration in these cases is not usually that important as we want it to be a quick change.

## Proposal
It would be useful if there was a short default duration for this command. Something like 5 or 15 minutes. It would function similarly to the `!mute` command where if no duration is given, it opts for a 1 hour duration.

### PR DESCRIPTION
**Pull request by doublevcodes**
### Relevant issues
Closes #1515 

### Description
Makes the `duration` parameter optional giving it a default value of 1 hour as suggested in the issue

### CODE REVIEW

File: bot/exts/moderation/infraction/superstarify.py [hunk 1]
```diff
@@ -134,6 +135,10 @@ async def superstarify(
         if await _utils.get_active_infraction(ctx, member, "superstar"):
             return
 
+        # Set the duration to 1 hour if none was provided
+        if not duration:
+            duration = datetime.datetime.now() + datetime.timedelta(hours=1)
```
Comment (RohanJnr) [inline bot/exts/moderation/infraction/superstarify.py:140]:
```suggestion
        # Set the duration to 1 hour if none was provided
        duration = duration or datetime.datetime.utcnow() + datetime.timedelta(hours=1)
```

**Review by RohanJnr — CHANGES_REQUESTED** [review]

File: bot/exts/moderation/infraction/superstarify.py (hunk 1 above)
Comment (RohanJnr) [reply bot/exts/moderation/infraction/superstarify.py:140]: also, it would be better to store the default duration as a constant

Comment (RohanJnr) [reply bot/exts/moderation/infraction/superstarify.py]: like
```python
DEFAULT_DURATION = 1
```
```python
duration = duration or datetime.datetime.utcnow() + datetime.timedelta(hours=DEFAULT_DURATION)
```
in the file, as a constant

**Comment by mbaruh** [comment]
I'm wondering if it would be better to do the same thing done to set the default duration of the mute command, meaning use the `Duration` converter to create the default time. This would ensure that if there is ever a change to the implementation of the converter, the behaviors won't diverge.

**Comment by RohanJnr** [comment]
> I'm wondering if it would be better to do the same thing done to set the default duration of the mute command, meaning use the `Duration` converter to create the default time. This would ensure that if there is ever a change to the implementation of the converter, the behaviors won't diverge.

I thought of doing that, but the Duration().converter(...) needs to parse and convert the string `1h` to a timedelta object first and then add the date and timedelta object.
Whereas here, we can directly add the date and timedelta object

**Comment by mbaruh** [comment]
> I thought of doing that, but the Duration().converter(...) needs to parse and convert the string `1h` to a timedelta object first and then add the date and timedelta object.
> Whereas here, we can directly add the date and timedelta object

True, but I don't really mind the extra operation. It's cheap and hidden anyway.

**Comment by RohanJnr** [comment]
> > I thought of doing that, but the Duration().converter(...) needs to parse and convert the string `1h` to a timedelta object first and then add the date and timedelta object.
> > Whereas here, we can directly add the date and timedelta object
> 
> True, but I don't really mind the extra operation. It's cheap and hidden anyway.

alright then, we can use the Duration() class directly

File: bot/exts/moderation/infraction/superstarify.py [hunk 2]
```diff
@@ -134,6 +135,9 @@ async def superstarify(
         if await _utils.get_active_infraction(ctx, member, "superstar"):
             return
 
+        # Set the duration to 1 hour if none was provided
+        duration = duration or datetime.datetime.utcnow() + datetime.timedelta(hours=1)
```
Comment (RohanJnr) [inline bot/exts/moderation/infraction/superstarify.py:139]:
```suggestion
        duration = duration or await Duration().convert(ctx, "1h")
```
You can have the `1h` as a constant at the top of the file.

Comment (RohanJnr) [reply bot/exts/moderation/infraction/superstarify.py:139]: make sure to import `Duration`

File: bot/exts/moderation/infraction/superstarify.py [hunk 3]
```diff
@@ -134,6 +135,9 @@ async def superstarify(
         if await _utils.get_active_infraction(ctx, member, "superstar"):
             return
 
+        # Set the duration to 1 hour if none was provided
```
Comment (mbaruh) [inline bot/exts/moderation/infraction/superstarify.py:138]: This comment is outdated
```suggestion
        # Set to default duration if none was provided
```

**Review by mbaruh — CHANGES_REQUESTED** [review]

**Review by mbaruh — APPROVED** [review]
Thanks for contributing!

**Pull request merged**